implementations/rust/ockam/ockam_api/src/logs/exporting_configuration.rs

```diff
@@ -131,48 +172,120 @@ impl Display for ExportingConfiguration {
+pub fn is_exporting_via_portal_set() -> ockam_core::Result<bool> {
+    get_env_with_default(OCKAM_OPENTELEMETRY_EXPORT_VIA_PORTAL, false)
+}
+
+/// Return true to display messages during the setup of the export
+pub fn is_export_debug_set() -> ockam_core::Result<bool> {
+    get_env_with_default(OCKAM_OPENTELEMETRY_EXPORT_DEBUG, false)
+}
+
+/// Print a debug statement if OCKAM_OPENTELEMETRY_EXPORT_DEBUG is true
+fn print_debug(message: impl Into<String>) {
+    if is_export_debug_set().unwrap_or(false) {
+        println!("{}", message.into());
+    }
+}
+
 /// Return ExportingEnabled::On if:
 ///
 /// - Exporting has not been deactivated by the user
 /// - The opentelemetry endpoint is accessible
 ///
 fn exporting_enabled(
-    quiet: bool,
+    endpoint: &OpenTelemetryEndpoint,
     connection_check_timeout: Duration,
 ) -> ockam_core::Result<ExportingEnabled> {
-    if !is_exporting_set()? {
-        Ok(ExportingEnabled::Off)
+    if is_endpoint_accessible(&endpoint.url(), connection_check_timeout) {
+        print_debug("Exporting is enabled");
+        Ok(ExportingEnabled::On)
     } else {
-        let endpoint = opentelemetry_endpoint()?;
-        if is_endpoint_accessible(endpoint.clone(), connection_check_timeout) {
-            Ok(ExportingEnabled::On)
-        } else {
-            if !quiet {
-                eprintln!("Exporting OpenTelemetry events is disabled because the OpenTelemetry collector endpoint at {} cannot be reached after {}ms", endpoint, connection_check_timeout.as_millis());
-                eprintln!("You can disable the export of OpenTelemetry events with: `export OCKAM_OPENTELEMETRY_EXPORT=false` to avoid this connection check.");
-            }
-            Ok(ExportingEnabled::Off)
-        }
+        let endpoint_kind = match endpoint {
+            OpenTelemetryEndpoint::HttpsEndpoint(_) => "OpenTelemetry collector endpoint",
+            OpenTelemetryEndpoint::PortalEndpoint(_) => "opentelemetry inlet",
+        };
+        print_debug(format!("Exporting OpenTelemetry events is disabled because the {} at {} cannot be reached after {}ms", endpoint_kind, endpoint.url(), connection_check_timeout.as_millis()));
+        print_debug("You can disable the export of OpenTelemetry events with: `export OCKAM_OPENTELEMETRY_EXPORT=false` to avoid this connection check.");
+
+        print_debug("Exporting is disabled");
+        Ok(ExportingEnabled::Off)
     }
 }
 
 /// Return true if the endpoint can be accessed with a TCP connection
-fn is_endpoint_accessible(url: Url, connection_check_timeout: Duration) -> bool {
+fn is_endpoint_accessible(url: &Url, connection_check_timeout: Duration) -> bool {
     match to_socket_addr(url) {
         Some(address) => {
-            std::net::TcpStream::connect_timeout(&address, connection_check_timeout).is_ok()
+            let retries = FibonacciBackoff::from_millis(100);
+            let mut total_time = Duration::from_secs(0);
```

Comment:
You could use `let now = Instant::now()` and `now.elapsed()` instead?

Reply:
Much better, thanks!